NDTensors/src/adapt.jl

```diff
@@ -36,5 +36,13 @@ to_vector_type(arraytype::Type{<:AbstractVector}) = arraytype
 to_vector_type(arraytype::Type{Array}) = Vector
 to_vector_type(arraytype::Type{Array{T}}) where {T} = Vector{T}
 
-set_eltype_if_unspecified(arraytype::Type{Vector{T}}, eltype::Type) where {T} = arraytype
-set_eltype_if_unspecified(arraytype::Type{Vector}, eltype::Type) = Vector{eltype}
+function set_eltype_if_unspecified(
+  arraytype::Type{<:AbstractArray{T,N}}, eltype::Type=default_eltype()
```

Comment:
I guess we don't need the type parameters `{T,N}` here.

Reply:
We need at least the T parameter for the dispatch system to choose between the two `set_eltype_if_unspecified` functions. Else it will always convert `arraytype` to `arraytype{eltype}`

Reply:
Right, good point.